### PR TITLE
Add a view helper for bulk options, checkboxes and cart controls.

### PR DESCRIPTION
**Pull request by EreMaijala**
Spun from #2929: stop carrying around the flags for bulk actions, checkboxes and cart controls. Encapsulates all the logic in a view helper. 

TODO
- [ ] Note BC breaks in changelog when merging (some code refactored from CombinedController into Combined\Options)

### CODE REVIEW

File: module/VuFind/src/VuFind/View/Helper/Root/SearchSettings.php [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+     * Check if bulk options are enabled
+     *
+     * @return bool
+     */
+    public function bulkOptionsEnabled()
+    {
+        if (!($this->config['Site']['showBulkOptions'] ?? false)) {
+            return false;
+        }
+        $options = $this->params->getOptions();
```
Comment (demiankatz) [inline module/VuFind/src/VuFind/View/Helper/Root/SearchSettings.php]: Maybe we should add a `protected getOptions()` method that throws an exception if `$this->params` is unset, so we avoid fatal errors if somebody invokes this improperly. And on a related note, does it definitely make sense to require a params object when we only use the options object? (I'm fine with leaving it as-is if you think that will be useful for future growth; but just asking!)

File: themes/bootstrap3/templates/search/bulk-action-buttons.phtml [hunk 2]
```diff
@@ -1,11 +1,16 @@
-<?php if (isset($this->showCheckboxes) && $this->showCheckboxes): ?>
+<?php
+  $showBulkOptions = $this->searchSettings($this->params)->bulkOptionsEnabled();
```
Comment (demiankatz) [inline themes/bootstrap3/templates/search/bulk-action-buttons.phtml]: Would it be cleaner to just set `$searchSettings = $this->searchSettings($this->params);` here and then call the appropriate methods in the checks below? Not sure that there's a lot of value in assigning to variables if they're only used in one place each.

File: module/VuFind/src/VuFind/Controller/CombinedController.php [hunk 3]
```diff
@@ -169,8 +165,6 @@ public function resultsAction()
             ->get(\VuFind\Search\Options\PluginManager::class);
         $config = $this->serviceLocator->get(\VuFind\Config\PluginManager::class)
             ->get('combined')->toArray();
-        $supportsCart = false;
```
Comment (demiankatz) [inline module/VuFind/src/VuFind/Controller/CombinedController.php:172]: I confess that I haven't tested to confirm, but I'm concerned that this bit might be broken by the refactoring. The combined search displays the top-level cart/bulk controls based on whether ANY of the columns support the cart. Unless this logic/check is built into the Combined Options class, I think it's getting lost. Did you check this already? If not, let me know if you need help testing.

**Review by demiankatz — CHANGES_REQUESTED** [review]
Thanks for this, @EreMaijala! Here are some initial thoughts based on a readthrough of the changes. I'll do hands-on testing after we finish discussing this round of feedback.

File: module/VuFind/src/VuFind/View/Helper/Root/SearchSettings.php (hunk 1 above)
Comment (EreMaijala) [reply module/VuFind/src/VuFind/View/Helper/Root/SearchSettings.php]: Good idea, added!

File: themes/bootstrap3/templates/search/bulk-action-buttons.phtml (hunk 2 above)
Comment (EreMaijala) [reply themes/bootstrap3/templates/search/bulk-action-buttons.phtml]: Yes, done!

File: module/VuFind/src/VuFind/Controller/CombinedController.php (hunk 3 above)
Comment (EreMaijala) [reply module/VuFind/src/VuFind/Controller/CombinedController.php:172]: You were right, Combined Options didn't return correct information. Now fixed.

**Review by demiankatz — APPROVED** [review]
Looks good to me now! Thanks, @EreMaijala!

**Pull request merged**